src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,100 @@
+import { getPostDetails } from '../../api/posts';
+import { Pos } from '../../types/pos';
 import { NewCommentForm } from '../NewCommentForm';
+import { deleteComment, getPostComments } from '../../api/comments';
 import './PostDetails.scss';
+import { Comment } from '../../types/comment';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+interface Props {
+  postId: string;
+}
+
+export const PostDetails: React.FC<Props> = ({ postId }) => {
+  const [selectedPost, setSelectedPost] = useState<Pos | null>(null);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [comsVisibility, setComsVisibility] = useState<boolean>(true);
+  const [rerender, setRerender] = useState(false);
+
+  async function loadPost() {
+    try {
+      const loaded = await getPostDetails(postId);
+
+      setSelectedPost(loaded);
+    } catch (error) {
+      // eslint-disable-next-line no-alert
+      alert(error);
+    }
+  }
+
+  async function loadComments() {
+    const comsFromServer = await getPostComments(postId);
+
+    setComments(comsFromServer);
+  }
+
+  useEffect(() => {
+    loadPost();
+    loadComments();
+  }, [postId, rerender]);
+
+  const comsVisTrigger = () => {
+    setComsVisibility(prevState => !prevState);
+  };
+
+  const deleteCommentHandler = async (commId: number) => {
+    await deleteComment(commId);
+    loadComments();
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>{selectedPost?.title}</p>
+      </section>
+
+      <section className="PostDetails__comments">
+        {comments.length === 0
+          ? null
+          : (
+            <button
+              type="button"
+              className="button"
+              onClick={comsVisTrigger}
+            >
+              Hide&#160;
+              {comments.length}
+              &#160;comments
+            </button>
+          )}
```

Comment:
```suggestion
        {comments.length && (
          <button
            type="button"
            className="button"
            onClick={comsVisTrigger}
          >
            Hide&#160;
            {comments.length}
            &#160;comments
          </button>
        )}
```

Reply:
so better compare length with 0 and do not complicate with ternary operator